docs/architecture/adr-001-mint-truncation-refactor.md

```diff
@@ -0,0 +1,336 @@
+# ADR 001: Mint Truncations Handling Refactor
+
+## Changelog
+
+* 22-08-2022: Initial Draft
+
+## Status
+
+Draft: <https://github.com/osmosis-labs/osmosis/pull/2342>
+
+## Abstract
+
+This ADR focuses on refactoring `x/mint` module to mitigate the discrepancies
+between the actual and the projected inflation amounts.
+
+Currently, we under-mint due to truncations. In the first year of operations,
+this happens to be approximately 2600 uosmo. As a result, we cannot reach the
```

Comment:
the number is that small? That is surprising

Reply:
Yes, I've explored the idea of removing the indexes completely to only keep the refactor for decoupling module accounts here:
https://github.com/osmosis-labs/osmosis/tree/roman/mint-refactor

However, I think the complexity overhead is not large enough relative to the precision benefit. Let me know what you think